src/v/raft/service.h

```diff
@@ -297,15 +300,16 @@ class service final : public raftgen_service {
             }
 
             auto shard = _shard_table.shard_for(r.meta.group);
-            if (!ret.shard_requests.contains(shard)) {
-                auto hbeats = ss::make_foreign(
-                  std::make_unique<std::vector<append_entries_request>>());
-                hbeats->push_back(std::move(r));
-                ret.shard_requests.emplace(shard, std::move(hbeats));
-                continue;
+            auto it = ret.shard_requests.find(shard);
+            if (it == ret.shard_requests.end()) {
+                auto result = ret.shard_requests.emplace(
+                  shard,
```

Comment:
nit: if you want to further reduce lookups then you can use the `hint` versions of emplace.

Reply:
great point